lib/ace/mode/c_cpp_highlight_rules.js

```diff
@@ -69,7 +69,7 @@ var c_cppHighlightRules = function() {
                 next : "comment"
             }, {
                 token : "string", // character
-                regex : "'(?:" + escapeRe + "|.)'"
+                regex : "['](?:" + escapeRe + "|.)*?[']"
```

Comment:
In c strings with single quote can have only one character/escape sequense, perhaps this should be 
```js
 "'(?:" + escapeRe + "|.)?'"
```
 instead?